sqlmesh/lsp/completions.py

```diff
@@ -97,3 +108,62 @@ def get_keywords_from_tokenizer(dialect: t.Optional[str] = None) -> t.Set[str]:
         parts = keyword.split(" ")
         expanded_keywords.update(parts)
     return expanded_keywords
+
+
+def get_dialect(context: t.Optional[LSPContext], file_uri: t.Optional[URI]) -> t.Optional[str]:
+    """
+    Get the dialect for a given file.
+    """
+    if file_uri is not None and context is not None and file_uri.to_path() in context.map:
+        file_info = context.map[file_uri.to_path()]
+
+        # Handle ModelInfo objects
+        if isinstance(file_info, ModelTarget) and file_info.names:
+            model_name = file_info.names[0]
+            model_from_context = context.context.get_model(model_name)
+            if model_from_context is not None and model_from_context.dialect:
+                return model_from_context.dialect
+
+        # Handle AuditInfo objects
+        elif isinstance(file_info, AuditTarget) and file_info.name:
+            audit = context.context.standalone_audits.get(file_info.name)
+            if audit is not None and audit.dialect:
+                return audit.dialect
+
+    if context is not None:
+        return context.context.default_dialect
+
+    return None
+
+
+def extract_keywords_from_content(content: str, dialect: t.Optional[str] = None) -> t.Set[str]:
+    """
+    Extract identifiers from SQL content using the tokenizer.
+    Only extracts identifiers (variable names, table names, column names, etc.)
+    that are not SQL keywords.
+    """
+    if not content:
+        return set()
+
+    tokenizer_class = Tokenizer
+    if dialect is not None:
+        try:
+            tokenizer_class = Dialect.get_or_raise(dialect).tokenizer_class
+        except Exception:
+            pass
+
+    keywords = set()
+    try:
+        tokenizer = tokenizer_class()
+        tokens = tokenizer.tokenize(content)
+        for token in tokens:
+            # Extract identifiers (variables, table names, column names, etc.)
+            if token.token_type == TokenType.VAR and token.text:
```

Comment:
What about `IDENTIFIER` (e.g. `"foo"`) and `STRING` (e.g. `'foo'`) tokens? Should we factor the set of interesting tokens out into a global variable and do a look up `token.token_type` in it instead?